README.md

```diff
@@ -377,126 +114,64 @@ workflow.add_edge("tools", 'agent')
 app = workflow.compile()
-```
-
-### Use it!
 
-We can now use it!
-This now exposes the [same interface](https://python.langchain.com/docs/expression_language/) as all other LangChain runnables.
-This [runnable](https://python.langchain.com/docs/expression_language/interface/) accepts a list of messages.
-
-```python
-from langchain_core.messages import HumanMessage
-
-inputs = {"messages": [HumanMessage(content="what is the weather in sf")]}
-app.invoke(inputs)
+# Use the Runnable
+final_state = app.invoke({"messages": [HumanMessage(content="what is the weather in sf")]})
+final_state["messages"][-1].content
 ```
 
-This may take a little bit - it's making a few calls behind the scenes.
-In order to start seeing some intermediate results as they happen, we can use streaming - see below for more information on that.
-
-## Streaming
-
-LangGraph has support for several different types of streaming.
-
-### Streaming Node Output
-
-One of the benefits of using LangGraph is that it is easy to stream output as it's produced by each node.
-
-```python
-inputs = {"messages": [HumanMessage(content="what is the weather in sf")]}
-for output in app.stream(inputs, stream_mode="updates"):
-    # stream() yields dictionaries with output keyed by node name
-    for key, value in output.items():
-        print(f"Output from node '{key}':")
-        print("---")
-        print(value)
-    print("\n---\n")
 ```
-
+'The current weather in San Francisco is as follows:\n- Temperature: 60.1°F (15.6°C)\n- Condition: Partly cloudy\n- Wind: 5.6 mph (9.0 kph) from SSW\n- Humidity: 83%\n- Visibility: 9.0 miles (16.0 km)\n- UV Index: 4.0\n\nFor more details, you can visit [Weather API](https://www.weatherapi.com/).'
 ```
-Output from node 'agent':
----
-{'messages': [AIMessage(content='', additional_kwargs={'function_call': {'arguments': '{\n  "query": "weather in San Francisco"\n}', 'name': 'tavily_search_results_json'}})]}
 
----
+### Step-by-step Breakdown:
 
-Output from node 'tools':
----
-{'messages': [FunctionMessage(content="[{'url': 'https://weatherspark.com/h/m/557/2024/1/Historical-Weather-in-January-2024-in-San-Francisco-California-United-States', 'content': 'January 2024 Weather History in San Francisco California, United States  Daily Precipitation in January 2024 in San Francisco Observed Weather in January 2024 in San Francisco  San Francisco Temperature History January 2024 Hourly Temperature in January 2024 in San Francisco  Hours of Daylight and Twilight in January 2024 in San FranciscoThis report shows the past weather for San Francisco, providing a weather history for January 2024. It features all historical weather data series we have available, including the San Francisco temperature history for January 2024. You can drill down from year to month and even day level reports by clicking on the graphs.'}]", name='tavily_search_results_json')]}
+1. <details>
+    <summary>Initialize the model, tools and the graph.</summary>
```

Comment:
you dont mention the graph in the bullet poitns here, but you mention in summaary

Reply:
shoudl mention Message state and what it is. this maybe worth separate item in list entirely?